### DISCUSSION TITLE
Using multiple or nested `Hono.prototype.route`

### DISCUSSION THREAD

**Discussion by kissge** [honojs/hono#945] [Q&A]
I've tried to use `route` like this, but couldn't make it work:

```ts
// api/photos/photos.routes.ts
const routes = new Hono().get('/list', (c) => c.jsonT({ photos: 0 }))
export default routes
```

```ts
// api/videos/videos.routes.ts
const routes = new Hono().get('/list', (c) => c.jsonT({ videos: 0 }))
export default routes
```

```ts
// api/routes.ts
import photos from './photos/photos.routes.ts'
import videos from './photos/photos.routes.ts'
const routes = new Hono().route('/photos', photos).route('/videos', videos)
export default routes
```

```ts
// api/index.ts
import routes from './routes'
const app = new Hono()
    .route('/', routes)
    .use(...)
    .use(...)
    .use(...)
```

Is this usage expected? Does hono support it? Any hint would be appreciated.

https://github.com/kissge/hono/commit/7e35ff5817b92c87a97276e6cea20edb57aa94c0

**Comment by usualoma** [comment]
Hi, @kissge.

Thank you for the report.

Your indicated usage is expected, but it was temporarily unavailable in v3.0.0. It has been fixed by #947 and is usable in v3.1.0.
I hope you will try it in v3.1.0.

Thank you,
